remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingAbstract.java

```diff
@@ -567,15 +567,12 @@ public void operationComplete(ChannelFuture f) throws Exception {
     }
 
     class NettyEventExecutor extends ServiceThread {
-        private final LinkedBlockingQueue<NettyEvent> eventQueue = new LinkedBlockingQueue<NettyEvent>();
         private final int maxSize = 10000;
+        private final LinkedBlockingQueue<NettyEvent> eventQueue = new LinkedBlockingQueue<NettyEvent>();
 
         public void putNettyEvent(final NettyEvent event) {
-            int currentSize = this.eventQueue.size();
-            if (currentSize <= maxSize) {
-                this.eventQueue.add(event);
-            } else {
-                log.warn("event queue size [{}] over the limit [{}], so drop this event {}", currentSize, maxSize, event.toString());
+            if (!eventQueue.offer(event)) {
```

Comment:
eventQueue is not bounded... it never fails...and this breaks the current trade-off impl I mentioned in the issue

Reply:
This comment: https://github.com/apache/rocketmq/issues/4533#issuecomment-1171878588